pymemcache/client/base.py

```diff
@@ -439,7 +440,14 @@ def close(self):
 
     disconnect_all = close
 
-    def set(self, key, value, expire=0, noreply=None, flags=None):
+    def set(
+        self,
+        key: KeyInputSpec,
+        value: Any,
```

Comment:
I'd like to have some constraints on `value`, but that might need to wait until we have more typing machinery in place for the serde interface.

Reply:
agreed